lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -211,6 +211,7 @@ export function useAPIFetch<T = unknown>(
   // something simpler, as long as it encodes the same information. The auth
   // token is not included in the key, as we assume currently that it does not
   // change the result.
-  const paramStr = params ? '?' + recordToSearchParams(params).toString() : '';
+  const queryString = recordToSearchParams(params ?? {}).toString();
+  const paramStr = queryString.length > 0 ? `?${queryString}` : '';
```

Comment:
We have two duplicated lines here and a few lines above. We could update `recordToSearchParams` to return a string instead of a URLSearchParams object, and take care of conditionally prepending the question mark, but I'll leave that out of this PR.